forc-pkg/src/source/git/auth.rs

```diff
@@ -0,0 +1,61 @@
+/// Handler holds all required information for handling authentication callbacks from `git2`.
+pub(crate) struct AuthHandler {
+    config: git2::Config,
+    /// Shows if the `AuthHandler` tried to make `SSH` authentication so far.
+    ssh_authentication_attempt: bool,
+    /// Shows if the `AuthHandler` tried to make `USER_PASS_PLAINTEXT` authentication so far.
+    plain_user_pass_attempt: bool,
+}
+
+impl AuthHandler {
+    /// Creates a new `AuthHandler` from all fields of the struct. If there are no specific reasons
+    /// not to, `default_with_config` should be prefered.
+    pub(crate) fn new(
```

Comment:
```suggestion
    fn new(
```

Let's make this private if it's not intended to be used elsewhere